test/API/H5_api_dataset_test.c

```diff
@@ -10453,9 +10459,12 @@ test_dataset_vlen_io(void)
     {
         H5Fclose(file_id);
         H5Gclose(container_group);
-        H5Dclose(dset_int);
-        H5Dclose(dset_float);
-        H5Dclose(dset_string);
+        if (dset_int != H5I_INVALID_HID)
+            H5Dclose(dset_int);
+        if (dset_float != H5I_INVALID_HID)
+            H5Dclose(dset_float);
+        if (dset_string != H5I_INVALID_HID)
+            H5Dclose(dset_string);
```

Comment:
Since these IDs were initialized to `H5I_INVALID_HID` and the purpose of the BEGIN_TRY and END_TRY blocks is to hide any errors while we try to close anything not yet closed while possibly closing stuff that IS closed, you shouldn't need to check for `H5I_INVALID_HID` here. That said, it doesn't hurt anything.